core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionPolicyManager.java

```diff
@@ -214,36 +214,36 @@ public Thread newThread(Runnable r) {
 
         SDKShutdownActivity.INSTANCE.register(this);
     }
-    
-	@Override
-	public void onTransmissionSent(TransmissionHandlerArgs transmissionArgs) {
-		for (TransmissionHandler handler : this.transmissionHandlers) {
-			handler.onTransmissionSent(transmissionArgs);
-		}
-	}
-	
-	@Override
-	public void addTransmissionHandler(TransmissionHandler handler) {
-		if(handler != null) {
-			this.transmissionHandlers.add(handler);
-		}
-	}
-	
-	/**
-	 * Set the number of retries before performing a back off operation.
-	 * @param maxInstantRetries Number of retries
-	 */
-	public void setMaxInstantRetries(int maxInstantRetries) {
-		if (maxInstantRetries >= 0 && maxInstantRetries < INSTANT_RETRY_MAX) {
-			INSTANT_RETRY_AMOUNT = maxInstantRetries;
-		}
-	}
-	
-	/**
-	 * Get the number of retries before performing a back off operation.
-	 * @return Number of retries
-	 */
-	public int getMaxInstantRetries() {
-		return INSTANT_RETRY_AMOUNT;
-	}
+
+    @Override
+    public void onTransmissionSent(TransmissionHandlerArgs transmissionArgs) {
+        for (TransmissionHandler handler : this.transmissionHandlers) {
+            handler.onTransmissionSent(transmissionArgs);
+        }
+    }
+
+    @Override
+    public void addTransmissionHandler(TransmissionHandler handler) {
+        if (handler != null) {
+            this.transmissionHandlers.add(handler);
+        }
+    }
+
+    /**
+     * Set the number of retries before performing a back off operation.
+     * @param maxInstantRetries Number of retries
+     */
+    public void setMaxInstantRetries(int maxInstantRetries) {
```

Comment:
I think we should always keep minimum instant retires to 3. Allowing instant retries to be set to 0 will again invite us for similar troubles in constrained networks. I would suggest that we change the condition to reflect that.

Reply:
instantRetries defaults to 3 if it is not set with this method, changed logic to only work if value is range [1..10]. I agree that 0 will put us in to a condition where we are backing off too soon, but disagree that we should always have this at 3 and give no option to lower.